Hangout-Bot/Code.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Responds to a MESSAGE event in Hangouts Chat.
+ *
+ * @param {Object} event the event object from Hangouts Chat
+ */
+function onMessage(event) {
+  var name = "";
+
+  if (event.space.type == "DM") {
+    name = "You";
+  } else {
+    name = event.user.displayName;
+  }
+  var message = name + " said \"" + event.message.text + "\"";
+
+  return { "text": message };
+}
+
+//map containing phases for DSA campaign creation.
+let phase_map = new Map([
+    [0, 'campaign_type'],
+    [1, 'sim_name'],
+    [2, 'start_date'],
+    [3, 'end_date'],
+    [3, 'budget'],
+    [4, 'country'],
+    [5, 'region'],
+    [6, 'domain'],
+    [7, 'target_page'],
+    [8, 'ad_text'],
+    [9, 'cpc']
+]);
+
+//boolean indicating if the creation phase has started.
+let creating = false;
```

Comment:
In general, if you need a comment to explain the variable, the variable name is not descriptive enough. Can you rename some of these variables (e.g. creating -> creationPhaseBegun)?

Reply:
Fixed